ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/TableSuppressBuilder.java

```diff
@@ -15,17 +15,98 @@
+import io.confluent.ksql.execution.builder.KsqlQueryBuilder;
+import io.confluent.ksql.execution.context.QueryContext;
 import io.confluent.ksql.execution.plan.KTableHolder;
-import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.execution.plan.KeySerdeFactory;
+import io.confluent.ksql.execution.plan.TableSuppress;
+import io.confluent.ksql.execution.streams.transform.KsTransformer;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.PhysicalSchema;
+import io.confluent.ksql.serde.SerdeOption;
+import java.util.Set;
+import java.util.function.BiFunction;
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.kstream.KTable;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.apache.kafka.streams.kstream.Suppressed;
+import org.apache.kafka.streams.state.KeyValueStore;
+
 
 public final class TableSuppressBuilder {
 
-  private TableSuppressBuilder() {
+  private static final String SUPPRESS_OP_NAME = "Suppress";
+
+  public TableSuppressBuilder() {
   }
 
-  public static <K> KTableHolder<K> build(
-      final KTableHolder<K> table
+  public <K> KTableHolder<K> build(
+      final KTableHolder<K> table,
+      final TableSuppress<K> step,
+      final KsqlQueryBuilder queryBuilder,
+      final KeySerdeFactory keySerdeFactory,
+      final MaterializedFactory materializedFactory
   ) {
-    throw new KsqlException("EMIT FINAL is not yet supported");
+    return build(
+        table,
+        step,
+        queryBuilder,
+        keySerdeFactory,
+        materializedFactory,
+        PhysicalSchema::from
+    );
+  }
+
+  @VisibleForTesting
+  @SuppressWarnings("unchecked")
+  <K> KTableHolder<K> build(
+      final KTableHolder<K> table,
+      final TableSuppress<K> step,
+      final KsqlQueryBuilder queryBuilder,
+      final KeySerdeFactory keySerdeFactory,
+      final MaterializedFactory materializedFactory,
+      final BiFunction<LogicalSchema, Set<SerdeOption>, PhysicalSchema> physicalSchemaFactory
+  ) {
+    final PhysicalSchema physicalSchema = physicalSchemaFactory.apply(
+        table.getSchema(),
+        step.getInternalFormats().getOptions()
+    );
+    final QueryContext queryContext = QueryContext.Stacker.of(
+        step.getProperties().getQueryContext())
+        .push(SUPPRESS_OP_NAME).getQueryContext();
+    final Serde<GenericRow> valueSerde = queryBuilder.buildValueSerde(
+        step.getInternalFormats().getValueFormat(),
+        physicalSchema,
+        queryContext
+    );
+    final Serde<K> keySerde = keySerdeFactory.buildKeySerde(
+        step.getInternalFormats().getKeyFormat(),
+        physicalSchema,
+        queryContext
+    );
+    final Materialized<K, GenericRow, KeyValueStore<Bytes, byte[]>> materialized =
+        materializedFactory.create(
+            keySerde,
+            valueSerde,
+            SUPPRESS_OP_NAME
+        );
```

Comment:
It looks like this is going to pass the name to Materialized and then pass the Materialized to transformValues, right? That's probably not what you want to do, since it will cause Streams to store a full extra copy of the table right before Suppress.

I'm guessing you just wanted to pass the key and value serdes into Suppress, in which case, you can replace this with:
```suggestion
    final Materialized<K, GenericRow, KeyValueStore<Bytes, byte[]>> materialized =
        Materialized.with(
            keySerde,
            valueSerde
        );
```

It's a subtle and dangerous API, I know.

Reply:
Ah great catch, thanks John!